src/Impostor.Api/Net/Messages/S2C/Message02StartGameS2C.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+using Impostor.Api.Games;
+
+namespace Impostor.Api.Net.Messages.S2C
+{
+    public static class Message02StartGameS2C
+    {
+        public static void Serialize(IMessageWriter writer, GameCode gameCode)
+        {
+            writer.StartMessage(MessageFlags.StartGame);
+            gameCode.Serialize(writer);
```

Comment:
MAJOR: I don't see the client reading out the game mode, although it does serialize it in InnerNetClient.SendStartGame

I suggest moving this code to the deserializing part of StartGameC2S and checking if the game code matches the currently connected game